### PR TITLE
Update token listing page

### PR DESCRIPTION
**Pull request by joel-medicala-yral**

### CODE REVIEW

File: ssr/src/component/buttons.rs [hunk 1]
```diff
@@ -0,0 +1,133 @@
+use leptos::*;
+
+#[component]
+pub fn Button(
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:4]: can we have a bit more descriptive name for this?

File: ssr/src/component/buttons.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    #[prop(optional)] disabled: Signal<bool>,
+) -> impl IntoView {
+    let on_click = move |_| on_click();
+    view! {
+        <button
+            on:click=on_click
+            disabled=disabled.get()
+            class=format!(
+                "w-full px-5 py-3 rounded-lg flex items-center transition-all justify-center gap-8 font-kumbh font-bold {} {}",
+                if disabled.get() {
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:18]: use the `disabled:` modifier from tailwind

File: ssr/src/component/buttons.rs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                    "text-white/50"
+                } else if alt_style.get() {
+                    "text-[#E2017B]"
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:21]: check color-primary-600 or color-primary-700

**Review by rupansh-sekar-yral — CHANGES_REQUESTED** [review]

File: ssr/src/component/buttons.rs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            style=format!(
+                "background: linear-gradient(73deg, {} );",
+                if disabled.get() {
+                    "#DE98BE 0%, #E761A9 33%, #7B5369 100%"
+                } else if alt_style.get() {
+                    "#FFF 0%, #FFF 1000%"
+                } else {
+                    "#DA539C 0%, #E2017B 33%, #5F0938 100%"
+                },
+            )
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:34]: These are not reactive

File: ssr/src/component/buttons.rs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+            style=format!(
+                "background: linear-gradient(73deg, {} );",
+                if disabled.get() {
+                    "#DE98BE 0%, #E761A9 33%, #7B5369 100%"
+                } else if alt_style.get() {
+                    "#FFF 0%, #FFF 1000%"
+                } else {
+                    "#DA539C 0%, #E2017B 33%, #5F0938 100%"
+                },
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:70]: Not reactive

File: ssr/src/component/buttons.rs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    children: Children,
+    disabled: Signal<bool>,
+    alt_style: Signal<bool>,
+    classes: String,
+    on_click: impl Fn() + 'static,
+) -> impl IntoView {
+    let on_click = move |_| on_click();
+    view! {
+        <button
+            disabled=disabled.get()
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:114]: this is not reactive

File: ssr/src/component/buttons.rs [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            on:click=on_click
+            class=format!(
+                "rounded-full border border-white text-sm font-bold font-kumbh px-5 py-2 {} {}",
+                if alt_style.get() {
+                    "bg-transparent text-white hover:bg-white/10 active:bg-white/5"
+                } else {
+                    "bg-white text-black"
+                },
+                classes,
```
Comment (rupansh-sekar-yral) [inline ssr/src/component/buttons.rs:123]: not reactive

File: ssr/src/page/icpump/mod.rs [hunk 8]
```diff
@@ -110,98 +55,302 @@ pub fn ICPumpListing() -> impl IntoView {
...
+                </ActionButton>
             </div>
         </div>
     }
 }
+
+#[component]
+pub fn PageSelector(page: RwSignal<u64>, end_of_list: RwSignal<bool>) -> impl IntoView {
+    view! {
+        <div class="flex gap-1 items-start text-sm font-medium text-[#A0A1A6]">
```
Comment (rupansh-sekar-yral) [inline ssr/src/page/icpump/mod.rs:220]: checkout color-zinc-400 or color-gray-400 or color-neutral-400

File: ssr/src/page/icpump/mod.rs [hunk 9]
```diff
@@ -110,98 +55,302 @@ pub fn ICPumpListing() -> impl IntoView {
...
+            <button
+                class="flex justify-center items-center w-8 h-8 rounded-lg bg-[#3A3A3A]"
```
Comment (rupansh-sekar-yral) [inline ssr/src/page/icpump/mod.rs:222]: color-neutral-900

File: ssr/src/page/icpump/mod.rs [hunk 10]
```diff
@@ -110,98 +55,302 @@ pub fn ICPumpListing() -> impl IntoView {
...
+pub fn ActionButtonWithHandler(
+    label: String,
+    children: Children,
+    #[prop(optional, default = false)] disabled: bool,
+    on_click: impl Fn() + 'static,
+) -> impl IntoView {
+    view! {
+        <button
+            disabled
+            on:click=move |_| {if !disabled{on_click()}}
```
Comment (rupansh-sekar-yral) [inline ssr/src/page/icpump/mod.rs:274]: ?

File: ssr/src/page/icpump/mod.rs [hunk 11]
```diff
@@ -110,98 +55,302 @@ pub fn ICPumpListing() -> impl IntoView {
...
+            class=format!("flex flex-col gap-1 justify-center items-center text-xs transition-colors {}", if !disabled{"group-hover:text-white text-[#D4D4D4]"}else{"group-hover:cursor-default text-[#525252]"})
```
Comment (rupansh-sekar-yral) [inline ssr/src/page/icpump/mod.rs:275]: `disabled:`

Comment (rupansh-sekar-yral) [reply ssr/src/page/icpump/mod.rs:275]: 525252 is color-neutral-600, d4d4d4 is color-neutral-300

**Review by rupansh-sekar-yral — CHANGES_REQUESTED** [review]
Can we make it a habit to use standard tailwind colors?
you can get most of the colors you need here: https://tailwindcss.com/docs/customizing-colors

or from the `color-primary` pallete that we have. Exact match may not be possible but its fine to use a slightly different but standard color

File: ssr/src/component/buttons.rs (hunk 4 above)
Comment (joel-medicala-yral) [reply ssr/src/component/buttons.rs:34]: Do they have to be reactive? beacause they dont change once rendered...

Comment (rupansh-sekar-yral) [reply ssr/src/component/buttons.rs]: You shouldn't use Signal if you don't want reactive behavior :P

**Review by rupansh-sekar-yral — APPROVED** [review]

**Pull request merged**